core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -735,27 +735,41 @@ public void setColumnComment(Session session, TableHandle tableHandle, ColumnHan
     }
 
     @Override
-    public void renameColumn(Session session, TableHandle tableHandle, ColumnHandle source, String target)
+    public void renameColumn(Session session, TableHandle tableHandle, QualifiedObjectName table, ColumnHandle source, String target)
     {
         CatalogHandle catalogHandle = tableHandle.getCatalogHandle();
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogHandle.getCatalogName());
```

Comment:
Hyper-nit - - since it's only used only in the if stm, I would have been tempted to move the `catalogMetadata` value into the if stm here and in the other two methods.

Reply:
I thought about that too. I'd rather keep it as-is though, personally; it matches the style of other methods that notify system security metadata.